sources/packages/backend/apps/api/Dockerfile

```diff
@@ -1,5 +1,5 @@
 # Base Image
-FROM artifacts.developer.gov.bc.ca/redhat-access-docker-remote/ubi8/nodejs-18:1-71.1697652955
+FROM artifacts.developer.gov.bc.ca/redhat-access-docker-remote/ubi8/nodejs-18:1-81
```

Comment:
While I support the node upgrade either way and the research looks great, we are still missing the root cause.
Can you please evaluate if the below assumptions make sense?

1 - Even the Openshift docker image defining a non-root user 1001, the container will be executed with a random user, as the error in the ticket also points out.
![image](https://github.com/bcgov/SIMS/assets/61259237/d40190c9-9124-4d8a-a6b3-c8f2a041572d)
The same is also supported by the below documentation.
https://docs.openshift.com/container-platform/4.11/openshift_images/create-images.html#use-uid_create-images

2 - Checking the BC git I found at least one entry applying the recommended solution due to a npm 9 issue.
![image](https://github.com/bcgov/SIMS/assets/61259237/a5bddaee-2fe2-4c7d-9b1c-e9f358b408e5)

The above would also explain why the error started when we moved from node 16 to 18 (npm 8 to 10).
Should we consider also applying the same?

Reply:
I see and I agree what @andrewsignori-aot about the OpenShift container running with  arbitrarily assigned user.

I am 100% on same page that we need to run a fix permission command to have the highest level of certainty that the issue is taken care at it's root cause.

When I look at the openshift, see that write permissions are not present outside the owner for the directory. 

![image](https://github.com/bcgov/SIMS/assets/54600590/ce753814-c6d8-4288-a114-384a053f1418)

 
There is one more thing which I want to share here. It may also be a passible solution.

Please go through this thread, https://github.com/sclorg/s2i-nodejs-container/issues/396

There is a mention about same error (we are using npm ci)

Following screenshots are the highlights linking to our issue

![image](https://github.com/bcgov/SIMS/assets/54600590/051f2e68-8b4c-4d43-a178-b6db65535371)

![image](https://github.com/bcgov/SIMS/assets/54600590/87116318-c6c1-4382-b1d0-9d31ccbc7f9d)



![image](https://github.com/bcgov/SIMS/assets/54600590/dd0c60d4-034b-425f-8485-466cef33ae5a)

Reply:
Thanks for the suggestions @andrewsignori-aot and @dheepak-aot , as suggested, i have updated the group permissions for user in group 0 to have write access to the folder./.npm and tested the build and deployments.

Reply:
Thanks for the further verifications @dheepak-aot 😉